server/cast/char.go

```diff
@@ -146,6 +146,20 @@ func charExplicit() {
 			return d, nil
 		},
 	})
+	framework.MustAddExplicitTypeCast(framework.TypeCast{
+		FromType: pgtypes.BpChar,
+		ToType:   pgtypes.Oid,
+		Function: func(ctx framework.Context, val any, targetType pgtypes.DoltgresType) (any, error) {
+			out, err := strconv.ParseInt(strings.TrimSpace(val.(string)), 10, 32)
```

Comment:
The best way to do this would be to parse it as a 64-bit integer and then do the range check. Parsing as a 32-bit integer will throw an error here, rather than further down.